### PR TITLE
Prac 09 feedback

### PR DESCRIPTION
**Pull request by AdriantoHartono**
Hi, can you guys help me to check the code?😅
@Lijiayi2
@JevanChandra
@a-rosa
@nora-zyy
@Riston-gitting-gud
@MikeLeongZhiFai

### CODE REVIEW

File: prac_09/my_band.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    kevin = Musician("Kevin Figueiredo")
+    band.add(kevin)
+
+    print("band (str)")
+    print(band)
+    print("band.play()")
+    print(band.play())
+
+
+main()
```
Comment (Lijiayi2) [inline prac_09/my_band.py:29]: all ok

File: prac_09/silver_service_taxi.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    def __init__(self, name, fuel, faciness):
+        super().__init__(name, fuel)
+        self.price_per_km *= faciness
+
+    def __str__(self):
+        return f"{super().__str__()} plus flagfall of ${self.flagfall:,.2f}"
+
+    def get_fare(self):
+        return super().get_fare() + self.flagfall
```
Comment (Lijiayi2) [inline prac_09/silver_service_taxi.py:18]: no problem

File: prac_09/silver_service_taxi_test.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+def main():
+    my_silver_taxi = SilverServiceTaxi("Humer", 200, 2)
+    print(my_silver_taxi)
+    my_silver_taxi.drive(18)
+    print(my_silver_taxi)
+    print(my_silver_taxi.get_fare())
+    my_silver_taxi.start_fare()
+    print(my_silver_taxi)
+
+main()
```
Comment (Lijiayi2) [inline prac_09/silver_service_taxi_test.py:15]: good code

File: prac_09/taxi.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    def start_fare(self):
+        """Begin a new fare."""
+        self.current_fare_distance = 0
+
+    def drive(self, distance):
+        """Drive like parent Car but calculate fare distance as well."""
+        distance_driven = super().drive(distance)
+        self.current_fare_distance += distance_driven
+        return distance_driven
+
```
Comment (Lijiayi2) [inline prac_09/taxi.py:37]: all good

File: prac_09/taxi_simulator.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    """Getting valid taxi input from the users"""
+    if current_taxi < 0 or current_taxi > len(taxis) - 1:
+        print("Invalid taxi choice")
+        taxi_chosen = None
+    else:
+        taxi_chosen = current_taxi
+    return taxi_chosen
+
+
+main()
```
Comment (Lijiayi2) [inline prac_09/taxi_simulator.py:58]: it's ok

File: prac_09/taxi_test.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Adrianto Hartono
+# 14347930
+
+from prac_09.taxi import Taxi
+
+def main():
+    my_taxi = Taxi("Prius 1", 100)
```
Comment (Lijiayi2) [inline prac_09/taxi_test.py:7]: I think you should add 1.23

File: prac_09/unreliable_car.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        super().__init__(name, fuel)
+        self.reliability = 0
+
+    def drive(self, distance):
+        random_number = random.randint(0,100)
+        if random_number < float(self.reliability):
+            driven_distance = super().drive(distance)
+        else:
+            driven_distance = 0
+        return driven_distance
```
Comment (Lijiayi2) [inline prac_09/unreliable_car.py:18]: good code

File: prac_09/unreliable_car_test.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Adrianto Hartono
+# 14347930
+
+from prac_09.unreliable_car import UnreliableCar
+
+def main():
+    my_unreliable = UnreliableCar("Prius 1", 100, 60)
+    my_unreliable.drive(40)
+    print(my_unreliable)
+
```
Comment (Lijiayi2) [inline prac_09/unreliable_car_test.py:10]: I think this code is not completed

Comment (AdriantoHartono) [reply prac_09/unreliable_car_test.py:10]: Thank you for the feedback, but I think that is all🙏

File: prac_09/unreliable_car.py (hunk 7 above)
Comment (AdriantoHartono) [reply prac_09/unreliable_car.py:18]: Thank you for the feedback 👌

File: prac_09/taxi_test.py (hunk 6 above)
Comment (AdriantoHartono) [reply prac_09/taxi_test.py:7]: Thank you for the feedback, but the 1.23 already imported from prac_09.taxi , hopefully it helps to answer your doubt🙏

File: prac_09/taxi_simulator.py (hunk 5 above)
Comment (AdriantoHartono) [reply prac_09/taxi_simulator.py:58]: Thank you for the feedback🙏

File: prac_09/taxi.py (hunk 4 above)
Comment (AdriantoHartono) [reply prac_09/taxi.py:37]: Thank you for the feedback🙏

File: prac_09/silver_service_taxi_test.py (hunk 3 above)
Comment (AdriantoHartono) [reply prac_09/silver_service_taxi_test.py:15]: Thank you for the feedback🙏

File: prac_09/silver_service_taxi.py (hunk 2 above)
Comment (AdriantoHartono) [reply prac_09/silver_service_taxi.py:18]: Thank you for the feedback🙏

File: prac_09/my_band.py (hunk 1 above)
Comment (AdriantoHartono) [reply prac_09/my_band.py:29]: Thank you for the feedback🙏

File: prac_09/band.py
Comment (JevanChandra) [inline prac_09/band.py:1]: Great code

File: prac_09/guitar.py
Comment (JevanChandra) [inline prac_09/guitar.py:1]: Nice work

File: prac_09/musician.py
Comment (JevanChandra) [inline prac_09/musician.py:1]: Great job

File: prac_09/my_band.py
Comment (JevanChandra) [inline prac_09/my_band.py:1]: Nice work

File: prac_09/silver_service_taxi.py
Comment (JevanChandra) [inline prac_09/silver_service_taxi.py:1]: Short and simple good job

File: prac_09/silver_service_taxi_test.py
Comment (JevanChandra) [inline prac_09/silver_service_taxi_test.py:1]: Nice work

File: prac_09/taxi.py
Comment (JevanChandra) [inline prac_09/taxi.py:1]: Good work figuring it out

File: prac_09/taxi_simulator.py
Comment (JevanChandra) [inline prac_09/taxi_simulator.py:1]: Nice

File: prac_09/taxi_test.py
Comment (JevanChandra) [inline prac_09/taxi_test.py:1]: Great solving skills

File: prac_09/taxi_simulator.py (hunk 5 above)
Comment (nora-zyy) [reply prac_09/taxi_simulator.py:58]: It was clearly completed as required.

File: prac_09/unreliable_car.py (hunk 7 above)
Comment (nora-zyy) [reply prac_09/unreliable_car.py:18]: The code writing approach is very clear.

File: prac_09/taxi_test.py
Comment (AdriantoHartono) [reply prac_09/taxi_test.py:1]: Thank you for the feedback🙏

File: prac_09/taxi_simulator.py
Comment (AdriantoHartono) [reply prac_09/taxi_simulator.py:1]: Thank you for the feedback🙏

File: prac_09/taxi.py
Comment (AdriantoHartono) [reply prac_09/taxi.py:1]: Thank you for the feedback🙏

File: prac_09/silver_service_taxi_test.py
Comment (AdriantoHartono) [reply prac_09/silver_service_taxi_test.py:1]: Thank you for the feedback🙏

File: prac_09/silver_service_taxi.py
Comment (AdriantoHartono) [reply prac_09/silver_service_taxi.py:1]: Thank you for the feedback🙏

File: prac_09/my_band.py
Comment (AdriantoHartono) [reply prac_09/my_band.py:1]: Thank you for the feedback🙏

File: prac_09/musician.py
Comment (AdriantoHartono) [reply prac_09/musician.py:1]: Thank you for the feedback🙏

File: prac_09/guitar.py
Comment (AdriantoHartono) [reply prac_09/guitar.py:1]: Thank you for the feedback🙏

File: prac_09/band.py
Comment (AdriantoHartono) [reply prac_09/band.py:1]: Thank you for the feedback🙏

File: prac_09/unreliable_car.py (hunk 7 above)
Comment (AdriantoHartono) [reply prac_09/unreliable_car.py:18]: Thank you for the feedback🙏

File: prac_09/unreliable_car_test.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Adrianto Hartono
+# 14347930
+
+from prac_09.unreliable_car import UnreliableCar
+
+def main():
+    my_unreliable = UnreliableCar("Prius 1", 100, 60)
+    my_unreliable.drive(40)
+    print(my_unreliable)
```
Comment (Riston-gitting-gud) [inline prac_09/unreliable_car_test.py:9]: Good code it works!

File: prac_09/band.py
Comment (MikeLeongZhiFai) [reply prac_09/band.py:1]: Great code! I took some references from it!

File: prac_09/unreliable_car_test.py (hunk 9 above)
Comment (AdriantoHartono) [reply prac_09/unreliable_car_test.py:9]: Thank you for the feedback 🙏

File: prac_09/band.py
Comment (AdriantoHartono) [reply prac_09/band.py:1]: Thank you for the feedback 🙏

File: prac_09/taxi_simulator.py (hunk 5 above)
Comment (AdriantoHartono) [reply prac_09/taxi_simulator.py:58]: Thank you for the feedback 🙏

File: prac_09/band.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    def add(self, musician):
+        self.musicians.append(musician)
+
+    def play(self):
+        for musician in self.musicians:
+            if not vars(musician)['instruments']:
+                print(f"{vars(musician)['name']} needs an instrument!")
+            else:
+                print(f"{vars(musician)['name']} is playing: {vars(musician)['instruments'][0]}")
+        return ""
```
Comment (a-rosa) [inline prac_09/band.py:22]: Missing docstring

File: prac_09/taxi_test.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# Adrianto Hartono
+# 14347930
+
+from prac_09.taxi import Taxi
```
Comment (a-rosa) [inline prac_09/taxi_test.py:4]: You can just directly reference taxi instead of prac_09.taxi since the taxi file is in the same file as this file

**Pull request merged**